site/assets/js/cookie-consent.js

```diff
@@ -0,0 +1,149 @@
+ * SPDX-License-Identifier: MIT
+ *
+ * Note: Banner visibility is controlled by Hugo production mode.
+ * This script only loads in production environments.
+ */
+
+(function() {
+    'use strict';
+
+    /**
+     * Simple check for localhost/development to prevent analytics pollution
+     * @returns {boolean} True if running on localhost or with custom port
+     */
+    function isLocalhost() {
+        return location.hostname.includes('localhost') ||
+               location.hostname === '127.0.0.1' ||
+               location.port !== '';
+    }
+
+    /**
+     * Inject Google Analytics script into the page
+     */
+    function injectGAScript() {
+        // Defense in depth: prevent analytics on localhost even in production builds
+        if (isLocalhost()) {
+            console.log('Analytics disabled for localhost');
+            return;
+        }
+
+        // Check if GA tracking ID is available
+        if (!window.GOOGLE_TAG_ID) {
+            console.log('GA tracking ID not configured');
+            return;
+        }
+
+        if (document.getElementById('ga-script')) return;
+
+        // Load the external Google Analytics script
+        var s = document.createElement('script');
+        s.id = 'ga-script';
+        s.async = true;
+        s.src = 'https://www.googletagmanager.com/gtag/js?id=' + window.GOOGLE_TAG_ID;
+        document.head.appendChild(s);
+
+        // Initialize Google Analytics directly
+        window.dataLayer = window.dataLayer || [];
+        window.gtag = function() { dataLayer.push(arguments); };
+        window.gtag('js', new Date());
+        window.gtag('config', window.GA_TRACKING_ID);
+    }
+
+    /**
+     * Clear all analytics-related cookies and data
+     */
+    function clearAnalyticsCookies() {
+        // Clear Google Analytics cookies
+        var cookies = document.cookie.split(';');
+        for (var i = 0; i < cookies.length; i++) {
+            var cookie = cookies[i].trim();
+            var cookieName = cookie.split('=')[0];
+
+            // Remove GA cookies
+            if (cookieName.startsWith('_ga') ||
+                cookieName.startsWith('_gid') ||
+                cookieName.startsWith('_gat')) {
+
+                // Clear for current domain
+                document.cookie = cookieName + '=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;';
+                // Clear for domain with leading dot
+                document.cookie = cookieName + '=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/; domain=' + location.hostname + ';';
+                document.cookie = cookieName + '=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/; domain=.' + location.hostname + ';';
+            }
+        }
+
+        // Clear dataLayer
+        if (window.dataLayer) {
+            window.dataLayer = [];
+        }
+
+        // Properly disable Google Analytics tracking
+        if (window.GOOGLE_TAG_ID) {
+            window['ga-disable-' + window.GOOGLE_TAG_ID] = true;
```

Comment:
What if the user disables cookies and then enables again? We should delete this property in `injectGAScript`.